### PR TITLE
Add searches controller

### PR DESCRIPTION
**Pull request by margaretoti**
This branch removes the responsibility of searching for books from the
books controller and creates a new `book_searches` controller to handle
all book related searches. Now when a user searches for a book or
author, she is directed to a new page where search results are listed.
Previously search results were listed in the `books_path`.

book_searches_controller

### CODE REVIEW

File: app/views/book_searches/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<%= link_to "New search", books_path %>
```
Comment (paulcsmith) [inline app/views/book_searches/index.html.erb]: This could probably be I18n

File: app/views/book_searches/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <li>
+      <%= book.title %>,
+      <%= book.author %>
+      <%= link_to t('books.index.check_out_link'), book_checkouts_path(book, current_user), method: :post %>
+    </li>
+  <% end %>
+</ul>
+
+<% if @book_results.blank? %>
+  <h4> <%= t('book_searches.index.no_results', param: params[:search]) %>.</h4>
```
Comment (paulcsmith) [inline app/views/book_searches/index.html.erb]: You could remove the space after the h4 tag

File: app/views/book_searches/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<%= link_to "New search", books_path %>
+<h3> <%= t('book_searches.index.header', param: params[:search]) %> </h3>
+
+<ul class="search_results">
+  <% @book_results.each do |book| %>
+    <li>
```
Comment (paulcsmith) [inline app/views/book_searches/index.html.erb:6]: If this is the same as the item rendered in the books index then you may want to extract a partial for it

**Review by paulcsmith — COMMENTED** [review]
A few small comments. Overall I like this direction!

File: app/views/book_searches/_book_search_bar.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<%= form_tag(book_search_path, method: :get, id: 'search_form') do %>
```
Comment (margaretoti) [inline app/views/book_searches/_book_search_bar.html.erb:1]: @paulcsmith, As a result of using a `book_searches` controller does this line change at all. I assumed I could now use `simple_form_for` and remove `method: :get` but if I do both of those things, the functionality breaks.

Comment (paulcsmith) [reply app/views/book_searches/_book_search_bar.html.erb:1]: You can probably use simple_form_for but you need method: :get so the form knows it needs to submit as a get still

**Pull request merged**